hw_11/HW_1_Sotnikov.py

```diff
@@ -0,0 +1,368 @@
+
+            chat.recieve(self)           # Передаем сообщение в чат
+
+# Creating class User
+class User:
+    def __init__(self, nickname, name="Bogdan", surname="Sotnikov", \
+                  birhdate="01.05.1999", sex="male",\
+                  user_status="user", what_is_he_doing="doing his python homework"):
+        self.name = name
+        self.surname = surname
+        self.nickname = nickname
+        self.birhdate = birhdate
+        self.sex = sex
+        self.user_status = user_status
+        self.what_is_he_doing = what_is_he_doing
+
+    def __repr__(self):
+        return str([self.nickname,
+                    self.name,
+                    self.surname,
+                    self.birhdate,
+                    self.sex,
+                    self.user_status,
+                    self.what_is_he_doing
+                   ])
+
+
+#2 Alien assignment
+class Args:
+    def __init__(self, *args, **kwargs):
+        self.argi = args
+        self.kwargi = kwargs
+
+    def __rlshift__(self, fun):
+        return fun(*self.argi, **self.kwargi)
+
+    def regeneration(self):
+        print("Восстановление нервных клеток")
+
+sum << Args([1, 2])
+
+(lambda a, b, c: a**2 + b + c) << Args(1, 2, c=50)
+
+
+#3 The successor
+
+class StrangeFloat(float):
+    def __init__(self, num):
+        super().__init__()
+        self.num = num
+        self.operands = {'add': lambda a, b: a + b, 'subtract': lambda a, b: a - b,
+                         'multiply': lambda a, b: a * b, 'divide': lambda a, b: a / b }
+    def __getattr__(self, attrname):
+        if "_" in attrname:
+            attrs = attrname.split("_")
+            attr = attrs[0]
+            value = float(attrs[1])
+            return type(self)(self.operands[attr](self, value))
+
+number = StrangeFloat(3.5)
+
+number.add_1
+
+number.subtract_20
+
+number.multiply_5
+
+number.divide_25
+
+number.add_1.add_2.multiply_6.divide_8.subtract_9
+
+getattr(number, "add_-2.5")   # Используем getattr, так как не можем написать number.add_-2.5 - это SyntaxError
+
+
+# 4 Dunders vs normals
+
+## Before
+
+import numpy as np
+
+
+matrix = []
+for idx in range(0, 100, 10):
+    matrix += [list(range(idx, idx + 10))]
+
+selected_columns_indices = list(filter(lambda x: x in range(1, 5, 2), range(len(matrix))))
+selected_columns = map(lambda x: [x[col] for col in selected_columns_indices], matrix)
+
+arr = np.array(list(selected_columns))
+
+mask = arr[:, 1] % 3 == 0
+new_arr = arr[mask]
+
+product = new_arr @ new_arr.T
+
+if (product[0] < 1000).all() and (product[2] > 1000).any():
+    print(product.mean())
+
+## After
+
+np = __import__("numpy")
+
+
+matrix = []
+for idx in range(0, 100, 10):
+    matrix.__iadd__([list(range(idx, idx.__add__(10)))])
+
+selected_columns_indices = list(filter(lambda x: x in range(1, 5, 2), range((matrix).__len__())))
+selected_columns = map(lambda x: [x.__getitem__(col) for col in selected_columns_indices], matrix)
```

Comment:
`__call__` можно поставить в куче мест
```suggestion
selected_columns = map.__call__(lambda x: [x.__getitem__(col) for col in selected_columns_indices], matrix)
```